src/commodore/inventory.ts

```diff
@@ -0,0 +1,63 @@
+import { execSync } from 'child_process';
+
+import { logger } from 'renovate/dist/logger';
+
+import { cacheDir, pruneObject, writeYamlFile } from './util';
+
+import type { CommodoreParameters, Facts } from './types';
+
+let versionCache: Map<string, CommodoreParameters> = new Map();
+
+function cacheKey(facts: Facts): string {
+  return `${facts.distribution}-${facts.cloud}-${facts.region}`;
+}
+
+export function clearCache(): void {
+  versionCache = new Map();
+}
+
+export async function writeFactsFile(
+  cacheKey: string,
+  facts: Facts
+): Promise<string> {
+  /* Write facts class for `commodore inventory component` */
+  const factsPath: string = `${cacheDir()}/${cacheKey}-facts.yaml`;
+  try {
+    await writeYamlFile(factsPath, {
+      parameters: { facts: pruneObject(facts) },
+    });
+  } catch (err) {
+    logger.error(`Error writing facts YAML: ${err}`);
+  }
+  return factsPath;
+}
+
+export async function renderInventory(
+  repoPath: string,
+  extraValuesPath: string,
+  facts: Facts
+): Promise<CommodoreParameters> {
+  const ck: string = cacheKey(facts);
+  let cachedVersions = versionCache.get(ck);
+  if (cachedVersions !== undefined) {
+    logger.info(`Reusing cached versions for ${ck}`);
+    return cachedVersions;
+  }
+
+  const factsPath: string = await writeFactsFile(ck, facts);
+
+  /* construct and execute Commodore command */
+  var command = `commodore inventory components ${repoPath} -ojson -f ${factsPath} -f ${extraValuesPath}`;
```

Comment:
I'm offended that it's the year 2021 and the abysmal node standard library still does not support passing arguments as an array.

Reply:
It does for `spawn()` (cf. https://nodejs.org/api/child_process.html#child_processspawncommand-args-options), but not for `exec()`.